### PR TITLE
applications: nrf5340_audio: Fix CIS and BIS for upmerge

### PR DESCRIPTION
**Pull request by alexsven**
- Added ext adv for BIS
- Changed callbacks for CIS
- Removed capabilities for CIS

Signed-off-by: Alexander Svensen <alexander.svensen@nordicsemi.no>

### CODE REVIEW

**Comment by desowin** [comment]
@rick1082 and @erikrobstad Could you please review?

File: applications/nrf5340_audio/src/bluetooth/Kconfig.defaults [hunk 1]
```diff
@@ -86,8 +86,7 @@ config BT_ASCS_ASE_SNK_COUNT
 	default 1
 
 config BT_ASCS_ASE_SRC_COUNT
-	default 1 if STREAM_BIDIRECTIONAL
-	default 0
+	default 1
```
Comment (desowin) [inline applications/nrf5340_audio/src/bluetooth/Kconfig.defaults]: Why?

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c [hunk 2]
```diff
@@ -87,7 +87,7 @@ static bool bitrate_check(const struct bt_codec *codec)
 
 static bool adv_data_parse(struct bt_data *data, void *user_data)
 {
-	if (data->type == BT_DATA_NAME_COMPLETE && data->data_len == DEVICE_NAME_PEER_LEN) {
+	if (data->type == 0x30 && data->data_len) {
```
Comment (desowin) [inline applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:90]: Where is this magic number from?

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c [hunk 3]
```diff
@@ -141,18 +143,109 @@ static struct bt_audio_stream_ops stream_ops = { .sent = stream_sent_cb,
...
+	if (ret) {
+		LOG_ERR("Unable to get broadcast ID: %d", ret);
+		return ret;
+	}
+
+	/* Setup extended advertising data */
+	net_buf_simple_add_le16(&ad_buf, BT_UUID_BROADCAST_AUDIO_VAL);
+	net_buf_simple_add_le24(&ad_buf, broadcast_id);
+
+	ext_ad[0].type = 0x30;
```
Comment (rick1082) [inline applications/nrf5340_audio/src/bluetooth/le_audio_bis_gateway.c:181]: We should use macro to define this.

File: applications/nrf5340_audio/src/bluetooth/Kconfig.defaults (hunk 1 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/Kconfig.defaults]: Due to a bug in host, fixed in https://github.com/zephyrproject-rtos/zephyr/pull/52026

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c (hunk 2 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:90]: It's the broadcast name number defined here: https://www.bluetooth.com/wp-content/uploads/2022/11/assigned_numbers_release.pdf, it hasn't been defined in host yet, but there is a PR for it (https://github.com/zephyrproject-rtos/zephyr/pull/52069)

File: applications/nrf5340_audio/src/bluetooth/Kconfig.defaults (hunk 1 above)
Comment (desowin) [reply applications/nrf5340_audio/src/bluetooth/Kconfig.defaults]: If I cherry-pick the fix as `[nrf fromtree]` then this change can be removed?

**Review by rick1082 — APPROVED** [review]

File: applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c (hunk 2 above)
Comment (desowin) [reply applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c:90]: If the PR is likely to be merged soon, then I don't see anything wrong with adding it as `[nrf fromtree]` and then using it here. Please update me on the PR.

File: applications/nrf5340_audio/src/bluetooth/Kconfig.defaults (hunk 1 above)
Comment (alexsven) [reply applications/nrf5340_audio/src/bluetooth/Kconfig.defaults]: yes :)

Comment (desowin) [reply applications/nrf5340_audio/src/bluetooth/Kconfig.defaults]: Both commits from the fix are now part of upmerge. Please update this PR.

**Comment by desowin** [comment]
@alexsven please just update the commit in your branch. It will show with conflicts here but I will handle it just fine. I put the not fully ready patch (this PR) into upstream branch mostly to have CI run on all patches I have received at once (as the patches are independent and cover different areas).

**Comment by desowin** [comment]
I should have probably created a separate branch, e.g. `upmerge-submissions` alongside `upmerge` branch. Then the PRs would go against `upmerge-submissions`. Then I could freely put things out in the `upmerge branch` (to trigger full CI) without confusing GitHub what is really going on. Doing so would allow doing actual per-patchset review without GitHub going crazy.

**Pull request closed without merging**